### ISSUE TITLE
Can't start SSH session

### ISSUE THREAD

**Issue by Fry-kun** [FreeRDP/Remmina#992]
* Client (OS name and version): Fedora 24
* Remmina version (remmina --version): 1.2.0-rcgit.7
* Desktop environment (GNOME, Unity, KDE, ..): XFCE
* Connecting to (OS and version): Fedora 23
* Connecting via (RDP, VNC, ...): SSH+VNC

This worked in Fedora 23, stopped working in 24
The error I'm getting is
```
Failed to startup SSH session: crypt_set_algorithm2: no crypto algorithm function found for chacha20-poly1305@openssh.com
```

Debug log:
```
[SSH] socket_callback_connected: Socket connection callback: 1 (0)
[SSH] ssh_client_connection_callback: SSH server banner: SSH-2.0-OpenSSH_7.2
[SSH] ssh_analyze_banner: Analyzing banner: SSH-2.0-OpenSSH_7.2
[SSH] ssh_analyze_banner: We are talking to an OpenSSH client version: 7.2 (70200)
[SSH] crypt_set_algorithms2: crypt_set_algorithms2: no crypto algorithm function found for chacha20-poly1305@openssh.com
```

client:
```
$ rpm -qa |grep libssh
libssh2-1.7.0-5.fc24.x86_64
libssh-0.7.3-1.fc24.x86_64
libssh2-1.7.0-5.fc24.i686
libssh2-devel-1.7.0-5.fc24.x86_64
$ rpm -qa |grep openssh
openssh-askpass-7.2p2-12.fc24.x86_64
openssh-7.2p2-12.fc24.x86_64
openssh-clients-7.2p2-12.fc24.x86_64
openssh-server-7.2p2-12.fc24.x86_64
```

client's /etc/ssh/ssh_config:
```
Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr
```

By experimenting:
* user ssh config is ignored; only system-wide version is used.
* if I remove all @openssh.com ciphers from the list (so that aes256-ctr is 1st), the connection completes without errors

So, the error seems to be that the libssh library (??) only attempts to use the 1st cipher in the list and doesn't try any of the others. Not sure if this is a problem with libssh calling convention or libssh itself (perhaps https://red.libssh.org/issues/202)

An ugly workaround for user is to adjust the ciphers -- but they'd have to change ciphers for the whole system!
An ugly workaround for Remmina: write a loop and whittle down the list of ciphers until the list is exhausted or connection is made. Python/pseudocode:
``` python
ciphers = read_config()    ## "chacha20-poly1305,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr"
for cipher in ciphers.split(','):
    # just guessing here 
    ssh_options_set (ssh->session, SSH_OPTIONS_CIPHERS_C_S, cipher);
    ssh_options_set (ssh->session, SSH_OPTIONS_CIPHERS_S_C, cipher);
    if connection_established():
        break
```

**Comment by antenore** [comment]
Yes, it's terrible! I prefer not working around the issue that way.

Another solution is to add the possibility to choose the cipher per Remmina profile, but it meas that it'll be up to the user to choose the best, therefore, this is also ugly. 

libssh development is now frozen since May. I'm reluctant taking decisions that will worsen the  Remmina code. 

I was considering the ideas to move to libssh2 or either use openssh (directly forking ssh process), but there will be a lot of work to do.

**Comment by Fry-kun** [comment]
If it's going to be that much work, the workaround loop hack would be preferable until "the right way" is implemented (shouldn't leave users hanging)
Looks like libssh2 doesn't support EC, though possibly some work being done on that..

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @Fry-kun
